File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/FreightTemplateService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbFreightTemplate;

import java.util.List;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface FreightTemplateService {

	/**
	 * 返回全部列表
	 * @return
	 */
    List<TbFreightTemplate> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
    PageResult findByPagination(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
    void add(TbFreightTemplate freightTemplate);
	
	
	/**
	 * 修改
	 */
    void update(TbFreightTemplate freightTemplate);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
    TbFreightTemplate findById(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
    void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
    PageResult findWithConditionsByPagination(TbFreightTemplate freightTemplate, int pageNum, int pageSize);
	
}

File: pinyougou-parent/pinyougou-shop-web/src/main/webapp/js/service/freightTemplateService.js
//服务层
app.service('freightTemplateService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../freightTemplate/findAll.php');
	};
	//分页 
	this.findByPagination=function(page,rows){
		return $http.get('../freightTemplate/findByPagination.php?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findOne=function(id){
		return $http.get('../freightTemplate/findOne.php?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../freightTemplate/add.php',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../freightTemplate/update.php',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../freightTemplate/delete.php?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../freightTemplate/findWithConditionsByPagination.php?page='+page+"&rows="+rows, searchEntity);
	}    	
});
